tests/test_graph.py

```diff
@@ -83,6 +84,41 @@ def test_invalid_recipe_order_and_bootstrap(names, bootstrap):
     assert "conflict" in e_info.value.message.lower()
 
 
+def test_blacklist():
+    # First, get order without blacklist:
+    build_order, python_modules, bs = get_recipe_order_and_bootstrap(
+        ctx, ["python3", "kivy"], None
+    )
+    # Now, obtain again with blacklist:
+    build_order_2, python_modules_2, bs_2 = get_recipe_order_and_bootstrap(
+        ctx, ["python3", "kivy"], None, blacklist=["libffi"]
+    )
+    assert "libffi" not in set(build_order_2)
```

Comment:
Minor: why enforcing the `set()` here? Creating the `set()` probably is `O(n)` anyway so why bothering?